### PR TITLE
Use PEP 503 instead of pypi xmlrpc to support mirrors (Fix #141)

### LINKED ISSUE #141: Setup: Provide an argument to use pypi mirror
Currently `index-url` is hardcoded in:
https://github.com/frida/frida-python/blob/86e52547a6c5702507b4e9323edd3195bb9fce80/src/setup.py#L80

It's painful when the pypi site is blocked under certain circumstances.

### PR DESCRIPTION
**Pull request by GalaxySnail**
Fix #141 

I couldn't test it on my environment, because I don't know what `FRIDA_EXTENSION` is. So I wrote a unittest for `PEP503PageParser`.

It is not easy to write a HTML parser, but I'm not sure if is it good to add a new third-party dependency to `pyproject.toml`.

(BTW, python2 is horrible for me. 😂)

(English is not my first language, please forgive my poor English.)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -44,18 +52,85 @@
     frida_extension = os.environ['FRIDA_EXTENSION']
 frida_major_version = int(frida_version.split(".")[0])
 
+Tag = namedtuple("Tag", ["tagname", "attrs"])
+ParsedUrlInfo = namedtuple("ParsedUrlInfo",
+                           ["url", "filename", "major", "minor", "micro"])
+
+
+def get_index_url_from_pip(pip):
+    cmd = [pip, "config", "get", "global.index-url"]
```
Comment (oleavr) [inline setup.py:61]: Could we do this through the API instead, or is `pip config` relying on private APIs?

**Review by oleavr — COMMENTED** [review]
Thanks, this looks great! 💥

File: setup.py (hunk 1 above)
Comment (GalaxySnail) [reply setup.py:61]: As far as I know, pip [doesn't provide any APIs](https://pip.pypa.io/en/stable/user_guide/#using-pip-from-your-program). The only interface it provides is the command line interface. The whole of pip's implementation is in its `_internal` package, which starts with an underline.

In addition, `[sys.executable, "-m", "pip", ...]` looks better than `["pip", ...]` here. (I forgot to read its doc yesterday.) In this way, `FRIDA_PIP` may become useless.

**Comment by GalaxySnail** [comment]
I think it's ready to be merged. ;-) @oleavr

File: setup.py (hunk 1 above)
Comment (oleavr) [reply setup.py]: Ok, makes sense, thanks for clarifying. (Sorry for the delay here btw!)

**Review by oleavr — APPROVED** [review]

**Comment by ChiChou** [comment]
Thank you very much!

**Pull request merged**